wordpress-mobile/WordPress-Android
# Refactor NotificationsDetailListFragment to kotlin

As part of the #14137 this PR moves the NotificationsDetailListFragment to kotlin.

The intention was only to move the fragment to kotlin, without affecting the feature, so the behaviour should have remained the same as before.

A few notes:
- Started the translation from AS auto conversion tool
- AsyncTask was kept for now and the update value (that seems to be not used) was moved from Boolean to Void
- again inside the AsyncTask, AS auto translated into a while loop the `for (int i = 0; i < bodyArray.length(); i++) {` and it seems consistent to me
- One thing I would like to get attention is the variable nullability moving to kotlin, sometimes could be tricky to determine if a variable should be managed as nullable or not, given that kotlin would crash the app if a null value ends up to be passed to a not nullable variable/parameter.

## To test
Having 2 Emulators with the develop branch and this one branch, compare the behaviour of the NotificationsDetailListFragment and check it behaves the same. Some scenarios I could think of:

- From Notifications select each of the filtered list type and open notification details 
- Open notification from automatcher
- Trigger a notification (like by commenting on a post) and open the notification details by tapping on the Android system notification
- Smoke test notifications approving/spamming/liking etc...better if having 2 emulators/devices to compare behaviours side by side

PR submission checklist:
